tests/acme_endorsement.py

```diff
@@ -0,0 +1,341 @@
+            }
+        }
+    }
+
+    for node in args.nodes:
+        endorsed_interface = infra.interfaces.RPCInterface(
+            host=infra.net.expand_localhost(),
+            endorsement=infra.interfaces.Endorsement(
+                authority=infra.interfaces.EndorsementAuthority.ACME
+            ),
+        )
+        endorsed_interface.public_host = network_name
+        endorsed_interface.acme_configuration = "pebble"
+        node.rpc_interfaces["acme_endorsed_interface"] = endorsed_interface
+        node.acme_challenge_server_interface = (
+            endorsed_interface.host + ":" + str(http_port)
+        )
+
+    exception_seen = None
+
+    with open(output_filename, "w", encoding="ascii") as out:
+        with open(error_filename, "w", encoding="ascii") as err:
+            mock_dns_proc = start_mock_dns(
+                mock_dns_filename,
+                mock_dns_listen_address,
+                mock_dns_mgmt_address,
+                out,
+                err,
+            )
+
+            register_endorsed_hosts(args, network_name, mock_dns_mgmt_address)
+
+            pebble_proc = start_pebble(
+                binary_filename,
+                config_filename,
+                mock_dns_listen_address,
+                out,
+                err,
+                env={"PEBBLE_WFE_NONCEREJECT": "0", "PEBBLE_VA_NOSLEEP": "1"},
+            )
+
+            try:
+                ca_certs = get_pebble_ca_certs(mgmt_address)
+                wait_for_certificates(args, network_name, ca_certs)
+            except Exception as ex:
+                exception_seen = ex
+            finally:
+                if pebble_proc:
+                    pebble_proc.kill()
+                if mock_dns_proc:
+                    mock_dns_proc.kill()
+
+    if exception_seen:
+        LOG.info("Pebble out:")
+        LOG.info(open(output_filename, "r", encoding="ascii").read())
+        LOG.info("Pebble err:")
+        LOG.info(open(error_filename, "r", encoding="ascii").read())
+        raise exception_seen
+
+
+@reqs.description("Test against Let's Encrypt's staging environment")
+def run_lets_encrypt(args):
+
+    # This requires a DNS name by which the network is reachable (which we don't have in the CI), e.g. your dev VM name.
+    # On the interface of that name, we also need port 80 to be reachable from the internet for the challenge responses (see Network on the Azure portal panel of your VM).
+    # The Let's Encrypt staging environment is described here: https://letsencrypt.org/docs/staging-environment/ (we need the CA certs for the staging environment, which are not globally endorsed).
+    # Further, to connect to Let's Encrypt, we also need their public root cert, which can be found at https://letsencrypt.org/certificates/
+    # (Clients won't need this as they usually have ISRG Root X1 installed, but our enclaves don't.)
+
+    service_dns_name = "acc-cwinter.uksouth.cloudapp.azure.com"
```

Comment:
I think it's better to avoid writing the name of your personal VM in this repo. Because this test needs some manual setup anyway, I'd rather have a comment here saying what this variable should be set to.